### ISSUE TITLE
Multiplayer Plugin - Too many objects are created for peers with containers

### ISSUE THREAD

**Issue by eleanorawesome** [Scirra/Construct-3-bugs#2410]
## Problem description

Trying to sync objects in a container makes too many
## Attach a .c3p

https://puu.sh/CdbjN/8a3bfa0a80.c3p
## Steps to reproduce

1.remote preview and connect with 2 or more players
2.
3.

## Observed result

there are more objects than there should be for peers
## Expected result

to have the same amount of objects for peers as the host
## Affected browsers/platforms

* Chrome: yes
* Firefox: yes

## System details

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 70.0.3538.110
Browser engine: Blink
Browser architecture: 64-bit
Context: webapp
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 12
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36
C3 release: r128 (beta)
Language setting: en-US

Local storage
Storage quota (approx): 16 gb
Storage usage (approx): 541 mb (3.3%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc.
Renderer: ANGLE (NVIDIA GeForce GTX 980 Ti Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_disjoint_timer_query_webgl2
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 96000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
</details>

**Comment by eleanorawesome** [comment]
https://github.com/Scirra/Construct-3-bugs/issues/2388 is sort of related but was created using families instead of containers for picking. Being unable to use either of these methods to sync variables at different rates forces expensive loops

**Comment by AshleyScirra** [comment]
Does this reproduce in both the C2 and C3 runtimes for you?

**Comment by eleanorawesome** [comment]
Yes this does reproduce in both the C2 and C3 runtimes

**Comment by AshleyScirra** [comment]
I've looked in to this and it's a very difficult problem. Firstly there is kind of a mismatch between the two features: networking is inherently unreliable, and different objects can be synced at different times, and created and destroyed at any time accordingly. Containers on the other hand have a hard requirement that all instances must be created and destroyed as one, in order for the internal algorithms in the event engine to work. Getting these to match up is a real challenge. Secondly supporting this would require changes to the networking protocol to support associating instances that were separately synced, and the feature was not designed to handle that, as it was written with the assumption synced objects can all work independently of one another.

Given the age of the issue and the complexity of the problem, I think it is best to just accept that this is not supported. I've updated the [manual entry](https://www.construct.net/en/make-games/manuals/construct-3/plugin-reference/multiplayer) to note that 'Sync object' does not support containers.
